src/hooks/useSettingsReducer.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useReducer, useState } from 'react';
+import { defaultSettings } from '../node/constants';
+import { encode } from '../node/encoding';
+import { get } from '../node/request';
+import { retrieveSettings, setSettings } from '../node/settings';
+import type { Settings } from '../node/types';
+import { Pin } from '../types';
+
+type Action =
+  | {
+      type: 'SET_SETTINGS';
+      payload: Settings;
+    }
```

Comment:
why is this action needed, remove

Reply:
needed for the fetch settings, to get them from file and put them in state